save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/LnkUserProjectController.kt

```diff
@@ -46,14 +46,52 @@ class LnkUserProjectController(
             ?: throw NoSuchElementException("There is no $projectName project in $organizationName organization")
         val usersWithRoles = if (projectPermissionEvaluator.hasPermission(authentication, project, Permission.READ)) {
             lnkUserProjectService.getAllUsersAndRolesByProject(project)
-                .filter { (_, role) -> role != Role.NONE }
+                .filter { (_, role) ->
+                    role != Role.NONE
+                }
                 .map { (user, role) ->
                     user.toUserInfo(mapOf(project.organization.name + "/" + project.name to role))
                 }
-                .also { logger.trace("Found ${it.size} users for project: $it") }
+                .also {
+                    logger.trace("Found ${it.size} users for project: $it")
+                }
         } else {
             emptyList()
         }
         return usersWithRoles
     }
+
+    /**
+     * @param organizationName
+     * @param projectName
+     * @param authentication
+     * @param prefix
+     * @return list of users, not connected to the project
+     * @throws NoSuchElementException
+     */
+    @GetMapping("/users/not-from/{organizationName}/{projectName}")
+    @Suppress("UnsafeCallOnNullableType")
+    fun getAllUsersNotFromProjectWithNamesStartingWith(
+        @PathVariable organizationName: String,
+        @PathVariable projectName: String,
+        @RequestParam prefix: String,
+        authentication: Authentication,
+    ): List<UserInfo> {
+        if (prefix.isEmpty()) {
+            return emptyList()
+        }
+        val project = projectService.findByNameAndOrganizationName(projectName, organizationName)
+            ?: throw NoSuchElementException("There is no $projectName project in $organizationName organization")
+        val projectUserIds = lnkUserProjectService.getAllUsersByProject(project).map { it.id!! }.toSet()
+        val exactMatchUsers = lnkUserProjectService.getNonProjectUsersByName(prefix, projectUserIds)
+        val prefixUsers = lnkUserProjectService.getNonProjectUsersByNamePrefix(
```

Comment:
Why do you need two requests? Is this because you want to display exact matches higher and `startsWith` will return results in mixed order? Please add some comments in code

Reply:
Yes, added comments